apps/studio/src/components/ResourceSelector/useResourceSelector.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback, useMemo } from "react"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+
+import type { ResourceItemContent } from "~/schemas/resource"
+import { isAllowedToHaveChildren } from "~/utils/resources"
+import { trpc } from "~/utils/trpc"
+import { lastResourceItemInAncestryStack } from "./utils"
+
+export const useResourceSelector = ({
+  interactionType,
+  siteId,
+  moveDest,
+  resourceStack,
+  isResourceHighlighted,
+  setIsResourceHighlighted,
+  existingResource,
+  setResourceStack,
+  removeFromStack,
+  onChange,
+}: {
+  interactionType: "move" | "link"
+  siteId: number
+  moveDest: ResourceItemContent | undefined
+  resourceStack: ResourceItemContent[]
+  isResourceHighlighted: boolean
+  setIsResourceHighlighted: (isResourceHighlighted: boolean) => void
+  existingResource: ResourceItemContent | undefined
+  setResourceStack: (resourceStack: ResourceItemContent[]) => void
+  removeFromStack: (numberOfResources: number) => void
```

Comment:
not sure if i'm missing context - why not let the client handle this since they can choose to call `useResourceStack`?

alternatively, why not just call `useResourceStack` inside this function? do we foresee these props changing (ie, not from `useResourceStack`?)